### DISCUSSION TITLE
Using the DatePicker inside of a React Vite project

### DISCUSSION THREAD

**Discussion by SamiKammoun** [tailwindlabs/tailwindcss#11372] [Help]
I just wanted to add the DatePicker component to my React TS project. It works smoothly without any errors. The only problem is that adding tailwind broke my existing work :c 
Setting preflight to false fixes my project but breaks the date picker.
I have also went as far as wrapping the component inside a div with an id and applying the preflight styles to all elements within it. It fixes the problem partially, the styling is back but it's not as it should be.

Did someone else run into this issue before ? I need help :(

**Comment by wongjn** [comment]
Please consider providing a [Tailwind Play](https://play.tailwindcss.com) or a remotely-hosted git repository that reproduces the undesired behavior.

**Reply by SamiKammoun** [reply]
[Here I have made a simulation](https://stackblitz.com/edit/vitejs-vite-kqtdva?file=tailwind.config.js),
If you set preflight to false, the project will be back to it's initial state (which is the expected old state of an existing project).
When preflight is set to true, the project is ruined.
I want the styling to only apply for the DatePicker component.

**Reply by wongjn** [reply]
You could consider disabling the core Tailwind preflight plugin and creating your own version of the preflight plugin that is scoped to your date picker DOM only.

**Reply by SamiKammoun** [reply]
I have been trying to do this using Shadow DOM. But it raises a few issues, the DatePicker element can no longer inherit the tailwind css elements, this way it will not be styled properly. Also, importing tailwind in the shadow dom scope is an antipattern that will eventually lead to performance issues (this is what I have found so far).
Is there another workaround ?

**Reply by wongjn** [reply]
You could consider reviewing the styles that Tailwind preflight applies to the date picker DOM, disable the core Tailwind preflight plugin, and then apply these styles to the date picker DOM with equivalent Tailwind classes.
